dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/deduplication/PotentialDuplicatesEnrollmentsTests.java

```diff
@@ -0,0 +1,233 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.tracker.deduplication;
+
+import com.google.gson.JsonObject;
+import org.hisp.dhis.ApiTest;
+import org.hisp.dhis.Constants;
+import org.hisp.dhis.actions.LoginActions;
+import org.hisp.dhis.actions.UserActions;
+import org.hisp.dhis.actions.UserRoleActions;
+import org.hisp.dhis.actions.tracker.PotentialDuplicatesActions;
+import org.hisp.dhis.actions.tracker.importer.TrackerActions;
+import org.hisp.dhis.dto.TrackerApiResponse;
+import org.hisp.dhis.helpers.JsonObjectBuilder;
+import org.hisp.dhis.helpers.file.JsonFileReader;
+import org.hisp.dhis.utils.DataGenerator;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * @author Gintare Vilkelyte <vilkelyte.gintare@gmail.com>
+ */
+public class PotentialDuplicatesEnrollmentsTests
+    extends ApiTest
+{
+    private final String userPassword = "Test1212?";
+
+    private PotentialDuplicatesActions potentialDuplicatesActions;
+
+    private UserActions userActions;
+
+    private TrackerActions trackerActions;
+
+    @BeforeEach
+    public void beforeEach()
+    {
+        potentialDuplicatesActions = new PotentialDuplicatesActions();
+        trackerActions = new TrackerActions();
+        userActions = new UserActions();
+
+        new LoginActions().loginAsSuperUser();
+    }
+
+    @Test
+    public void shouldNotBeMergedWhenBothTeisEnrolledInSameProgram()
+    {
+        String teiA = createTeiWithEnrollmentsAndEvents( Constants.TRACKER_PROGRAM_ID, "nlXNK4b7LVr" ).extractImportedTeis()
+            .get( 0 );
```

Comment:
Could we extract uid as constant so we have more visibility? Also, I was thinking what if tomorrow, uid generation changes?

Reply:
You mean the program stage id? It's not going to change, as it's hard-coded in the metadata setup file we import before tests. We can extract it, of course, but having it as constant might be a bit confusing. We could have a program model with program stages and have that as a constant.